### PR TITLE
fix: return mail and userPrincipalName for GroupMembership user members

### PR DESCRIPTION
**Pull request by ytsarev**
### Description of your changes

Expanded group members are deserialized into typed SDK objects, so user properties (mail, userPrincipalName) and service principal appId live on the typed getters rather than in additionalData. Read from the typed getters with an additionalData fallback, and request the fields explicitly via a nested $select. Adds direct processMember tests covering typed users, service principals, and the DirectoryObject fallback path.

Fixes #115

render test

```
crossplane render xr.yaml group-membership-example.yaml functions.yaml \
  --function-credentials=./secrets/azure-creds.yaml -rc
..
  groupMembers:
  - displayName: Yury Tsarev
    id: <omitted>
    type: user
    userPrincipalName: yury@upbound.io
```

I have:

- [x] Read and followed Crossplane's [contribution process].
- [x] Added or updated unit tests for my change.

[contribution process]: https://git.io/fj2m9
[docs]: https://docs.crossplane.io/contribute/contribute

### CODE REVIEW

File: fn.go [hunk 1]
```diff
@@ -576,7 +576,10 @@ func (g *GraphQuery) fetchGroupMembers(ctx context.Context, client *msgraphsdk.G
 	// See: https://developer.microsoft.com/en-us/graph/known-issues/?search=25984
 	requestConfig := &groups.GroupItemRequestBuilderGetRequestConfiguration{
 		QueryParameters: &groups.GroupItemRequestBuilderGetQueryParameters{
-			Expand: []string{"members"},
+			// Explicitly select the standard member fields via a nested $select so
+			// that user properties such as mail and userPrincipalName are returned
+			// for the expanded members (see issue #115).
+			Expand: []string{"members($select=id,displayName,mail,userPrincipalName,appId)"},
```
Comment (stevendborrelli) [inline fn.go:582]: Expand: []string{"members($select=id,displayName,mail,userPrincipalName,appId)"},

  The SDK needs @odata.type to deserialize members into typed objects (Userable,
  ServicePrincipalable). OData specifies that @odata.type is a system metadata annotation
  returned by default even when $select is present, and the manual test in the PR
  description confirms this works. However, if the service ever stops returning it
  implicitly, everything silently falls back to plain DirectoryObject and the typed path
  breaks. Consider adding @odata.type explicitly to be defensive:

  Expand:
  []string{"members($select=id,displayName,mail,userPrincipalName,appId,@odata.type)"},

File: fn_test.go [hunk 2]
```diff
@@ -3815,3 +3816,91 @@ func TestIdentityType(t *testing.T) {
 		})
 	}
 }
+
+// newTestUser builds a typed user directory object, mirroring how the Graph SDK
```
Comment (stevendborrelli) [inline fn_test.go:3820]: Missing edge-case test: typed user with empty mail

  The current test uses a user with both mail and userPrincipalName set. There's no test
  for a typed Userable where GetMail() returns nil or "" — in that case the code correctly
   skips setting mail, but a test would make that invariant explicit and prevent future
  regressions.

**Review by stevendborrelli — DISMISSED** [review]
LGTM, a couple of comments from a LLM scan.

File: fn.go (hunk 1 above)
Comment (ytsarev) [reply fn.go:582]: Thanks! I looked into this and decided to keep `$select` as-is, for two reasons:

1. **`@odata.type` is already returned by default** for heterogeneous `directoryObject` collections — it's how Graph disambiguates member types (the docs note that even for objects the caller lacks permission to read, "only the `@odata.type` ... and the id" come back). It isn't a normal selectable property, so adding it to `$select` risks a `400` that would break the whole query — strictly worse than today.

2. **The feared failure mode is already handled gracefully.** If members ever did deserialize as plain `DirectoryObject` (no type annotation), the selected fields land in `additionalData`, and `extractUserProperties`/`extractServicePrincipalProperties` fall back to reading from there — so the data is not silently lost. That path is covered by the `PlainDirectoryObjectUsesAdditionalDataFallback` test.

I also added a `TypedUserWithoutMailOmitsMail` test per your other comment to lock in the empty-`mail` behavior.

File: fn_test.go (hunk 2 above)
Comment (ytsarev) [reply fn_test.go:3820]: added, thanks!

**Review by stevendborrelli — APPROVED** [review]
